flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/batch/table/validation/CalcValidationTest.scala

```diff
@@ -128,4 +128,96 @@ class CalcValidationTest extends TableTestBase {
 
     table.select('a.flatten(), 'a.flatten())
   }
+
+  @Test
+  def testAddColumns(): Unit = {
+    val util = batchTestUtil()
+    val tab = util.addTable[(Int, Long, String)]("Table3",'a, 'b, 'c)
+
+    // Test aggregates
+    try {
+      tab.addColumns('a.sum)
+      fail("TableException expected")
+    } catch {
+      case _: TableException => //ignore
+    }
+
+    // Test replace the existing column, but do not with the alias.
+    try {
+      tab.addColumns(true, concat('c, "Sunny"))
+      fail("TableException expected")
+    } catch {
+      case _: TableException => //ignore
+    }
+
+  }
+
+  @Test
+  def testRenameColumns(): Unit = {
+    val util = batchTestUtil()
+    val tab = util.addTable[(Int, Long, String)]("Table3",'a, 'b, 'c)
+
+    // Test aggregates
+    try {
```

Comment:
Separate those test cases. Use `expected=Exception` annotation.

Reply:
I find that some of the test case using `expected=Exception` and some of the test case using `try/cach`, so if test multi-case I like using `try/catch`, but I am fine if you want to change it into `expected=Exception`. Do you also prefer using `expected=Exception`?